DigitalLearningSolutions.Data/Services/EmailService.cs

```diff
@@ -0,0 +1,127 @@
+                || mailSenderAddress == null
+            )
+            {
+                var errorMessage = GenerateConfigValueMissingMessage(
+                    mailServerUsername,
+                    mailServerPassword,
+                    mailServerAddress,
+                    mailServerPortString
+                );
+                throw new ConfigValueMissingException(errorMessage);
+            }
+
+            var mailServerPort = int.Parse(mailServerPortString);
+
+            MimeMessage message = CreateMessage(email, mailSenderAddress);
+
+            try
+            {
+                using var client = smtpClientFactory.GetSmtpClient();
+                client.Timeout = 10000;
+                client.Connect(mailServerAddress, mailServerPort);
+
+                client.Authenticate(mailServerUsername, mailServerPassword);
+
+                client.Send(message);
+                client.Disconnect(true);
+            }
+            catch 
+            {
+            }
+        }
+
+        private MimeMessage CreateMessage(Email email, string mailSenderAddress)
+        {
+            var message = new MimeMessage();
+            message.From.Add(MailboxAddress.Parse(mailSenderAddress));
+            foreach (string toAddress in email.To)
+            {
+                message.To.Add(MailboxAddress.Parse(toAddress));
+            }
+            foreach (string ccAddress in email.Cc)
+            {
+                message.Cc.Add(MailboxAddress.Parse(ccAddress));
+            }
+            foreach (string bccAddress in email.Bcc)
+            {
+                message.Bcc.Add(MailboxAddress.Parse(bccAddress));
+            }
+            message.Subject = email.Subject;
+            message.Body = email.Body.ToMessageBody();
+
+            return message;
+        }
+
+        private static string GenerateConfigValueMissingMessage(
+            string? mailServerUsername,
+            string? mailServerPassword,
+            string? mailServerAddress,
+            string? mailServerPortString
+            )
+        {
+            if (mailServerUsername == null)
+            {
+                return "Encountered an error while trying to send an email: The value of mailserverUsername is null";
+            }
+
+            if (mailServerPassword == null)
+            {
+                return "Encountered an error while trying to send an email: The value of mailserverPassword is null";
+            }
+
+            if (mailServerAddress == null)
+            {
+                return "Encountered an error while trying to send an email: The value of mailServerAddress is null";
+            }
+
+            if (mailServerPortString == null)
+            {
+                return "Encountered an error while trying to send an email: The value of mailServerPortString is null";
+            }
+
+            return "Encountered an error while trying to send an email: The value of mailSenderAddress is null";
```

Comment:
I appreciate this is copied from somewhere else, but do you think it's worth having a private method for generating this error string that takes in a name of the value that's null? Then, if we ever want to modify the error message, we only need to do it in one place. :)